### PR TITLE
New hardware definition added (mainly CPUs)

### PR DESCRIPTION
**Pull request by jouvin**

### CODE REVIEW

File: hardware/cpu/amd/epyc_7302.pan [hunk 1]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/amd/epyc_7302;
+
+"manufacturer" = "AMD";
+"model" = "AMD EPYC 3GHz, 16C/32T";
+"speed" = 3000; # MHz
+"arch" = "x86_64";
+"cores" = 16;
+"max_threads" = 32;
+"type" = "zen"; # AMD codename
```
Comment (jrha) [inline hardware/cpu/amd/epyc_7302.pan:9]:
```suggestion
"type" = "zen2"; # AMD codename
```

File: hardware/cpu/amd/epyc_7402.pan [hunk 2]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/amd/epyc_7402;
+
+"manufacturer" = "AMD";
+"model" = "AMD EPYC 3GHz, 24C/48T";
+"speed" = 2800; # MHz
+"arch" = "x86_64";
+"cores" = 24;
+"max_threads" = 48;
+"type" = "zen"; # AMD codename
```
Comment (jrha) [inline hardware/cpu/amd/epyc_7402.pan:9]:
```suggestion
"type" = "zen2"; # AMD codename
```

File: hardware/cpu/amd/epyc_7302.pan [hunk 3]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/amd/epyc_7302;
+
+"manufacturer" = "AMD";
+"model" = "AMD EPYC 3GHz, 16C/32T";
```
Comment (jrha) [inline hardware/cpu/amd/epyc_7302.pan:4]:
```suggestion
"model" = "AMD EPYC 7302 16-Core Processor";
```
Or at least, that's how it appears in `lscpu` and `/proc/cpuinfo`

File: hardware/cpu/amd/epyc_7402.pan [hunk 4]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/amd/epyc_7402;
+
+"manufacturer" = "AMD";
+"model" = "AMD EPYC 3GHz, 24C/48T";
```
Comment (jrha) [inline hardware/cpu/amd/epyc_7402.pan:4]:
```suggestion
"model" = "AMD EPYC 7402 24-Core Processor";
```
Or at least, that's how it appears in `lscpu` and `/proc/cpuinfo`.

File: hardware/cpu/intel/xeon_phi_3120a.pan [hunk 5]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/intel/xeon_phi_3120a;
+
+"manufacturer" = "Intel";
+"model" = "Intel(R) Xeon Phi Coprocessor 3120A @ 1.10 GHz";
+"speed" = 1100; # MHz
+"arch" = "x86_64";
+"cores" = 57;
+"max_threads" = 57;
+"type" = "Knights Corner"; # Intel codename
+"power" = 300; # TDP in watts
```
Comment (jrha) [inline hardware/cpu/intel/xeon_phi_3120a.pan:10]: Are you actually using these as system CPUs and managing an OS on them with Quattor?

File: hardware/cpu/intel/xeon_silver_4208.pan [hunk 6]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/intel/xeon_silver_4208;
+
+"manufacturer" = "Intel";
+"model" = "Intel(R) Xeon(R) Silver 4208 CPU @ 2.10 GHz";
+"speed" = 2100; # MHz
+"arch" = "x86_64";
+"cores" = 8;
+"max_threads" = 16;
+"type" = "skylake"; # Intel codename
```
Comment (jrha) [inline hardware/cpu/intel/xeon_silver_4208.pan:9]:
```suggestion
"type" = "cascade lake"; # Intel codename
```
https://ark.intel.com/content/www/us/en/ark/products/193390/intel-xeon-silver-4208-processor-11m-cache-2-10-ghz.html

File: hardware/cpu/intel/xeon_silver_4210R.pan [hunk 7]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/intel/xeon_silver_4210R;
+
+"manufacturer" = "Intel";
+"model" = "Intel(R) Xeon(R) Silver 4210R CPU @ 2.40 GHz";
+"speed" = 2400; # MHz
+"arch" = "x86_64";
+"cores" = 20;
+"max_threads" = 40;
+"type" = "skylake"; # Intel codename
```
Comment (jrha) [inline hardware/cpu/intel/xeon_silver_4210R.pan:9]:
```suggestion
"type" = "cascade lake"; # Intel codename
```
https://ark.intel.com/content/www/us/en/ark/products/197098/intel-xeon-silver-4210r-processor-13-75m-cache-2-40-ghz.html

File: hardware/cpu/intel/xeon_silver_4214Y.pan [hunk 8]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/intel/xeon_silver_4214Y;
+
+"manufacturer" = "Intel";
+"model" = "Intel(R) Xeon(R) Silver 4214Y CPU @ 2.20 GHz";
+"speed" = 2200; # MHz
+"arch" = "x86_64";
+"cores" = 12;
+"max_threads" = 24;
+"type" = "skylake"; # Intel codename
```
Comment (jrha) [inline hardware/cpu/intel/xeon_silver_4214Y.pan:9]:
```suggestion
"type" = "cascade lake"; # Intel codename
```
https://ark.intel.com/content/www/us/en/ark/products/193553/intel-xeon-silver-4214y-processor-16-5m-cache-2-20-ghz.html

**Review by jrha — COMMENTED** [review]
Are you actually using Knights Corner Phi's as system CPUs and managing an OS on them with Quattor? I was under the impression that this was only supported with Knights Landing. Otherwise these are coprocessors and should live under `hardware/card`.

**Comment by jouvin** [comment]
> Are you actually using Knights Corner Phi's as system CPUs and managing an OS on them with Quattor? I was under the impression that this was only supported with Knights Landing. Otherwise these are coprocessors and should live under `hardware/card`.

@jrha honnestly, I just pushed what we had in our database without checking if it was used. From a quick `grep`, I'd say that they are not used, not sure why they were added and who did that (nothing explicit in our change history).

If you want, I can remove the Knights Corner related templates from the PR. Let me know.

**Comment by jrha** [comment]
If they're not used then please remove them, I don't suppose anyone will suddenly need Knights Corner in the future.

**Comment by jouvin** [comment]
Done.

File: hardware/cpu/intel/xeon_gold_5218r.pan [hunk 9]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/cpu/intel/xeon_gold_5218r;
+
+"manufacturer" = "Intel";
+"model" = "Intel(R) Xeon(R) Gold 5218R CPU @ 2.r130 GHz";
```
Comment (jrha) [inline hardware/cpu/intel/xeon_gold_5218r.pan:4]: Looks like a typo crept in here.
```suggestion
"model" = "Intel(R) Xeon(R) Gold 5218R CPU @ 2.10 GHz";
```

**Review by jrha — APPROVED** [review]

**Pull request merged**